### DISCUSSION TITLE
How to set BT audio sink to NullStream

### DISCUSSION THREAD

**Discussion by unpapardo** [pschatzmann/arduino-audio-tools#1573] [Q&A] [answered]
Hello

I'm not sure if I'm doing something wrong, so I'm not opening an issue about this
I'm trying to add status beeps (i.e. when connecting or disconnecting a phone), and the previous suggestion was to

> This should be quite easy. I would suggest the following:
> 
> Make sure that A2DP stops any output (if necessary)
> call avrc pause()
> set the output to a NullStream
> copy the status beep signal to the SPDIFOutput. You can e.g. use a [SineGenerator for this](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-stream/streams-generator-spdif/streams-generator-spdif.ino).
> Resume A2DP Playback
> set output back to the SPDIFOutput
> call avrc resume()
> Alternatively you can use an [approach using the callback method](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-communication/a2dp/basic-a2dp-i2s/basic-a2dp-i2s.ino), where you can add some status variable to manage if you want to output the received signal or your own generated one

However, when setting the bt sink to a NullStream, nothing changes, the sound just keeps coming. I can copy the audio from a soundGenerator into the analog output, but both audio sources overlay and its not pretty. 
Here's a basic sketch that I would expect to change between sound and silence every few seconds, but it always outputs audio insted.

```
#include "AudioTools.h"
#include "BluetoothA2DPSink.h"

bool snd_bt = false;

AudioInfo info_stereo(44100, 2, 16);

AnalogAudioStream out;
NullStream null_out;

BluetoothA2DPSink bt(out);

void setup(void) {
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Warning);

  auto config = out.defaultConfig(TX_MODE);
  config.copyFrom(info_stereo);
  out.begin(config);

  bt.set_auto_reconnect(true);
  bt.start("ESP32-music");
}

void loop() {
  if (millis() % 5000 < 2500) {
    if (!snd_bt){
      Serial.println(snd_bt);
      bt.set_output(out);
      snd_bt = true;
    }
  }
  if (millis() % 5000 > 2500) {
    if (snd_bt){
      Serial.println(snd_bt);
      bt.set_output(null_out);
      snd_bt = false;
    }
  }
}

```

Any ideas?
Cheers

**Comment by pschatzmann** [comment, marked as answer]
I committed [some corrections to the A2DP library](https://github.com/pschatzmann/ESP32-A2DP/commit/e1caae968077ef0afae7a59e5ec57ddabd75b826) that will address your issue

**Reply by unpapardo** [reply]
Yep, that seems to solve the issue
Once again, thanks for the response and for the effort
